common/utils.go

```diff
@@ -32,13 +32,10 @@ func GetTopic(sig string) ethcommon.Hash {
 	return b
 }
 
-// MakeDir makes a directory
+// MakeDir creates a directory, including leading directories, if they don't already exist.
+// File permissions of created directories are only granted to the current user.
 func MakeDir(dir string) error {
-	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
```

Comment:
`os.ModePerm` is a mask, so we want to avoid using it as a value which sets very open permissions. This is a great site if you ever need help with the octal values:
https://chmod-calculator.com/